apps/src/lib/kits/maker/boards/microBit/MBFirmataWrapper.js

```diff
@@ -42,13 +39,40 @@ export default class MicrobitFirmataWrapper extends MBFirmataClient {
     } else {
       // Use the given port. Assume the port has been opened by the caller.
       // This branch is for WebSerial pathway - only opening/closing port for now.
-      this.myPort = port;
+      this.myPort = port; // port is a WebSerialPortWrapper
+      this.serialPortWebSerial = port.port;
       this.myPort.on('data', this.dataReceived.bind(this));
+      this.requestFirmataVersion();
+      this.requestFirmwareVersion();
 
       return Promise.resolve();
     }
   }
 
+  async disconnect() {
+    // Close and discard the serial port.
+    if (!isWebSerialPort(this.myPort)) {
```

Comment:
Can you remind me - is the current plan to only ever support MB with WebSerial? Or is there an intermediate step where we will support it on the 'old path'?

Reply:
I believe we are currently supporting MB via Maker App right now so that curriculum can connect with the MB as they write the Physical Computing module for the MB.